src/transforms/workflow.ts

```diff
@@ -33,42 +37,91 @@ function mapStatus(status: string, conclusion?: string): string {
 	}
 }
 
-function mapPullRequests(pull_requests: GitHubPullRequest[]): JiraPullRequest[] {
-	return pull_requests.map(pr => (
-		{
-			commit: {
-				id: pr.head.sha,
-				repositoryUri: pr.head.repo.url,
-			},
-			ref: {
-				name: pr.head.ref,
-				uri: `${pr.head.repo.url}/tree/${pr.head.ref}`,
-			},
-		}
-	));
+function mapPullRequests(
+	pull_requests: GitHubPullRequest[]
+): JiraPullRequest[] {
+	return pull_requests.map((pr) => ({
+		commit: {
+			id: pr.head.sha,
+			repositoryUri: pr.head.repo.url,
+		},
+		ref: {
+			name: pr.head.ref,
+			uri: `${pr.head.repo.url}/tree/${pr.head.ref}`,
+		},
+	}));
 }
 
-export default (context: Context): JiraBuildData | undefined => {
-	const { workflow_run, workflow } = context.payload;
-	const issueKeys = issueKeyParser().parse(`${workflow_run.head_branch}\n${workflow_run.head_commit.message}`);
+export default async (
+	githubClient: GitHubAPI,
+	payload: WorkflowPayload,
+	jiraHost: string,
+	logger?: LoggerWithTarget
+): Promise<JiraBuildData | undefined> => {
+	const { workflow_run, workflow } = payload;
+
+	const {
+		conclusion,
+		head_branch,
+		head_commit,
+		html_url,
+		name,
+		pull_requests,
+		repository,
+		run_number,
+		status,
+		updated_at,
+	} = workflow_run;
+
+	let issueKeys;
+
+	const supportBranchAndMergeWorkflowForBuildsFlagIsOn = await booleanFlag(BooleanFlags.SUPPORT_BRANCH_AND_MERGE_WORKFLOWS_FOR_BUILDS, false, jiraHost);
+	const workflowHasPullRequest = pull_requests.length > 0;
```

Comment:
Should it be under feature flag, too? (e.g. undefined `pull_requests` might blow the thing up)

Reply:
It shouldn't. `pull_requests` is always in the workflow run object. It's either an empty array if no pull request has been opened, or an array that contains 2 objects (head and base branches in the pr) if one has.

Reply:
Just realised there is already a comment there that says `// Optional information that links PRs. Min items: 1, Max items: 5`

Reply:
Yeah, I understand it shouldn't, but IMHO it is best to keep all the new code under a feature flag, especially if this is easy to achieve...

Reply:
I agree. However, in this case, pull_requests.length was already being used for the `references` in the payload. I just extracted it to a variable because it's now being used in 2 places in the code. If I only enable it if the FF is enabled, it means it will not be set to pass to the payload.